x-pack/plugins/security_solution/public/common/components/response_actions/response_actions_results.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { map } from 'lodash';
+import React from 'react';
+import type { EcsSecurityExtension as Ecs } from '@kbn/securitysolution-ecs';
+import { EndpointResponseActionResults } from './endpoint_action_results';
+import type { LogsEndpointAction } from '../../../../common/endpoint/types';
+import type { LogsOsqueryAction } from '../../../../common/search_strategy/security_solution/response_actions/actions';
+import { useKibana } from '../../lib/kibana';
+
+interface ResponseActionsResultsProps {
+  actions: Array<LogsEndpointAction | LogsOsqueryAction>;
+  ruleName?: string[];
+  ecsData: Ecs;
+}
+
+export const ResponseActionsResults = ({
+  actions,
+  ruleName,
+  ecsData,
+}: ResponseActionsResultsProps) => {
+  const {
+    services: { osquery, application },
+  } = useKibana();
+  const { OsqueryResult } = osquery;
+  const canReadOsquery = !!application?.capabilities?.osquery?.read;
+
+  return (
+    <>
+      {map(actions, (action) => {
+        if (isOsquery(action)) {
+          const actionId = action.action_id;
+          const queryId = action.queries[0].id;
+          const startDate = action['@timestamp'];
+
+          return (
+            <OsqueryResult
+              key={actionId}
+              actionId={actionId}
+              queryId={queryId}
+              startDate={startDate}
+              ruleName={ruleName}
+              ecsData={ecsData}
+              canReadOsquery={canReadOsquery}
+            />
+          );
+        }
+        if (isEndpoint(action)) {
+          return <EndpointResponseActionResults action={action} />;
+        }
```

Comment:
Thanks Ash, good point! You are right that it's either osquery or endpoint, however for the safety reasons I decided to have a check for isEndpoint before I start rendering the components just in case there is some data missing / some unwanted action pops by. Are you ok with leaving this as it is?